skbio/io/format/stockholm.py

```diff
@@ -0,0 +1,324 @@
+
+    #=GS O83071/259-312 AC O83071
+
+Example taken from [2]_.
+
+GC
+--
+Data relating to the columns of the multiple sequence alignment as a whole.
+Starts with #=GC followed by a feature and data relating to the feature.
+Typically comes at the end of the multiple sequence alignment.
+For example:
+
+.. code-block:: none
+
+    #=GC SS_cons            CCCCCHHHHHHHHHHHHH..EEEEEEEE....EEEEEEEEEEH
+
+Example taken from [2]_.
+
+GR
+--
+Data relating to the columns of a specific sequence in a multiple sequence
+alignment. Starts with #=GR followed by the sequence name followed by a feature
+and data relating to the feature. Typically comes after the data line it
+relates to.
+For example:
+
+.. code-block:: none
+
+    #=GR O31698/18-71 SS    CCCHHHHHHHHHHHHHHH..EEEEEEEE....EEEEEEEEHHH
+
+Example taken from [2]_.
+
+Examples
+========
+Suppose we have a Stockholm file:
+
+>>> import skbio.io
+>>> from io import StringIO
+>>> from skbio import RNA, TabularMSA
+>>> fs = '\\n'.join([
+...         '# STOCKHOLM 1.0',
+...         '#=GF RA    Deiman BA, Kortlever RM, Pleij CW;',
+...         '#=GF RL    J Virol 1997;71:5990-5996.',
+...         'AF035635.1/619-641             UGAGUUCUCGAUCUCUAAAAUCG',
+...         'M24804.1/82-104                UGAGUUCUCUAUCUCUAAAAUCG',
+...         'J04373.1/6212-6234             UAAGUUCUCGAUCUUUAAAAUCG',
+...         'M24803.1/1-23                  UAAGUUCUCGAUCUCUAAAAUCG',
+...         '#=GC SS_cons                   .AAA....<<<<aaa....>>>>',
+...         '//'
+...     ])
+>>> fh = StringIO(fs)
+>>> msa = skbio.io.read(fh, into=TabularMSA, constructor=RNA)
+>>> msa # doctest: +NORMALIZE_WHITESPACE
+TabularMSA[RNA]
+-------------------------------------------------
+Metadata:
+'RA': '   Deiman BA, Kortlever RM, Pleij CW;'
+'RL': '   J Virol 1997;71:5990-5996.'
+Positional metadata:
+'SS_cons': <dtype: object>
+Stats:
+sequence count: 4
+position count: 23
+-------------------------------------------------
+UGAGUUCUCGAUCUCUAAAAUCG
+UGAGUUCUCUAUCUCUAAAAUCG
+UAAGUUCUCGAUCUUUAAAAUCG
+UAAGUUCUCGAUCUCUAAAAUCG
+
+References
+==========
+.. [1] https://en.wikipedia.org/wiki/Stockholm_format
+.. [2] http://sonnhammer.sbc.su.se/Stockholm.html
+"""
+
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+from collections import OrderedDict, namedtuple
+
+from skbio.alignment import TabularMSA
+from skbio.sequence import Protein
+from skbio.io import create_format, StockholmFormatError
+
+stockholm = create_format('stockholm')
+_SeqData = namedtuple("SeqData", ["seq", "metadata", "pos_metadata"])
```

Comment:
Can you add a comment describing this data structure? Also, `positional_metadata` might be a better field name to match the `Sequence` constructor.